sysutils/salt/Portfile

```diff
@@ -29,31 +29,78 @@ if {$subport eq $name} {
-
     notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api."
-
-    variant python34 conflicts python35 python36 description {experimental python-3.4 support} {
+    
+    default_variants	+python27 +zmq
+    
+    variant python27 conflicts python34 python35 python36 description {experimental python-3.5 support} {
+		python.default_version 27
+		depends_build	    port:py${python.version}-setuptools
+	
+		depends_lib-append  port:py${python.version}-yaml \
+							port:py${python.version}-jinja2 \
+							port:py${python.version}-msgpack
+							
+		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api."
+    }
+    
+    variant python34 conflicts python27 python35 python36 description {experimental python-3.4 support} {
 		python.default_version 34
-		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. Support for Python 3 versions is experimental."
+		depends_build	    port:py${python.version}-setuptools
+	
+		depends_lib-append  port:py${python.version}-yaml \
+							port:py${python.version}-jinja2 \
+							port:py${python.version}-msgpack
+							
+		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. " \
+				"Support for Python 3 versions is experimental."
     }
-    variant python35 conflicts python34 python36 description {experimental python-3.5 support} {
+    
+    variant python35 conflicts python27 python34 python36 description {experimental python-3.5 support} {
 		python.default_version 35
-		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. Support for Python 3 versions is experimental."
+		depends_build	    port:py${python.version}-setuptools
+	
+		depends_lib-append  port:py${python.version}-yaml \
+							port:py${python.version}-jinja2 \
+							port:py${python.version}-msgpack
+							
+		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. " \
+				"Support for Python 3 versions is experimental."
     }
-    variant python36 conflicts python34 python35 description {experimental python-3.6 support} {
+    
+    variant python36 conflicts python27 python34 python35 description {experimental python-3.6 support} {
 		python.default_version 36
-		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. Support for Python 3 versions is experimental."
+		depends_build	    port:py${python.version}-setuptools
+	
+		depends_lib-append  port:py${python.version}-yaml \
+							port:py${python.version}-jinja2 \
+							port:py${python.version}-msgpack
+							
+		notes   "Salt startupitems are installed by subports salt-minion, salt-master, salt-syndic, salt-api. " \
+				"Support for Python 3 versions is experimental."
+    }
+    
+    variant zmq description {0mq TCPls  transport support} {
```

Comment:
Are the added dependencies that large that you cannot just enable all of them by default? Having a lot of variants usually leads to situations where the default installation does not necessarily work for all users and increases the testing complexity.

We usually try to avoid variants unless they are necessary.